test/lifecycle_test.dart

```diff
@@ -467,13 +519,20 @@ void sharedLifecycleTests<T extends react.Component>({
                 ? matchCall('shouldComponentUpdate', args: [expectedProps, initialState], state: initialState)
                 : matchCall('shouldComponentUpdateWithContext',
                     args: [expectedProps, initialState, newContext], state: initialState),
-            matchCall('componentWillUpdate', args: [expectedProps, initialState], state: initialState),
+            !isComponent2
+                ? matchCall('componentWillUpdate', args: [expectedProps, initialState], state: initialState)
+                : null,
             skipLegacyContextTests
                 ? null
                 : matchCall('componentWillUpdateWithContext',
                     args: [expectedProps, initialState, newContext], state: initialState),
             matchCall('render', state: initialState),
-            matchCall('componentDidUpdate', args: [expectedProps, initialState], state: initialState),
+            isComponent2
+                ? matchCall('getSnapshotBeforeUpdate', args: [expectedProps, initialState], state: initialState)
+                : null,
+            isComponent2
+                ? matchCall('componentDidUpdate', args: [expectedProps, initialState, null], state: initialState)
```

Comment:
#nit should be `snapshot`
```suggestion
                ? matchCall('componentDidUpdate', args: [expectedProps, initialState, snapshot], state: initialState)
```